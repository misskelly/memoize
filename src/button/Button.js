import React, {Component} from 'react';

export default class Button extends Component {

  handleClick = (e) => {
    const { selectDeck, removeFromDeck, getRandomCard, updateDeck, allCards, cardId } = this.props;
    
    switch (e.target.id) {
      case 'basicsDeckBtn':
        selectDeck('basics');
        break;
      case 'testingDeckBtn':
        selectDeck('testing');
        break;
      case 'removeBtn':
        removeFromDeck(cardId);
        break
      default:
        updateDeck(allCards)
      }
      
    getRandomCard();
  }
  render() {
    const { type, buttonText, label } = this.props;
    return (
      <div className='cardBtnWrapper'>
        <button className='cardBtn' 
                id={type}
                onClick={this.handleClick}>
          {buttonText}
        </button>
        <label  htmlFor={type} 
                className='btnLabel'>
          {label}
        </label>
      </div>
    )
  }

}