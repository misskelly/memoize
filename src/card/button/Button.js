import React, {Component} from 'react';

export default class Button extends Component {

  handleClick = (e) => {
    if (e.target.id === 'removeBtn') {
      this.props.removeFromDeck(this.props.cardId)
    }
  }
  render() {
    return (
      <>
        <button className='cardBtn' 
                id={this.props.type}
                onClick={this.handleClick}
        >
          {this.props.buttonText}
        </button>
        <label for={this.props.type} className='btnLabel'>
          {this.props.lable}
        </label>
      </>
    )
  }

}