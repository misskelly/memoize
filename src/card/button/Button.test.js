import React from 'react';
import Button from './Button';
import { shallow } from 'enzyme';

describe('Button', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(
      <Button />
    )
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

});