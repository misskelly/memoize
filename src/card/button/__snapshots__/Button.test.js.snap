// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Button />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <button
          className="cardBtn"
          onClick={[Function]}
        />,
        <label
          className="btnLabel"
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
          "className": "cardBtn",
          "id": undefined,
          "onClick": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
          "className": "btnLabel",
          "for": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "label",
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <button
            className="cardBtn"
            onClick={[Function]}
          />,
          <label
            className="btnLabel"
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
            "className": "cardBtn",
            "id": undefined,
            "onClick": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
            "className": "btnLabel",
            "for": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "label",
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
